src/Products/Viewer/Cache/ExternalResourcesStreamFactory.cs

```diff
@@ -0,0 +1,41 @@
+﻿using GroupDocs.Viewer.Interfaces;
+using GroupDocs.Viewer.Results;
+using System.IO;
+
+namespace GroupDocs.Viewer.MVC.Products.Viewer.Cache
+{
+    public class ExternalResourcesStreamFactory : IPageStreamFactory, IResourceStreamFactory
+    {
+        // Resources storage path.
+        private readonly string _outputPath;
+
+        // URL for resources download.
+        private readonly string _urlPrefix;
+
+        public ExternalResourcesStreamFactory(string cachePath, string resourcesPath)
+        {
+            if (!Directory.Exists(cachePath))
```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5933705)